src/patterns/dosdp-patterns/length_from_to.yaml

```diff
@@ -0,0 +1,103 @@
+---
+pattern_name: length_from_to
+
+pattern_iri: http://purl.obolibrary.org/obo/oba/patterns/bone_length_from_to.yaml
+
+description: 'A length trait that is the distance between two anatomical
+  reference points measured in a straight line.'
+
+#  examples:
+#    - http://purl.obolibrary.org/obo/FOVT_0001070  # femur length from caput to lateral condyle
+#    - http://purl.obolibrary.org/obo/FOVT_0001077  # length of talus along the midline
+#    - http://purl.obolibrary.org/obo/FOVT_0000007  # femur length from caput to medial condyle
+
+contributors:
+  - https://orcid.org/0000-0002-7073-9172  # David Osumi-Sutherland
+  - https://orcid.org/0000-0001-8314-2140  # Ray Stefancsik
+  - https://orcid.org/0000-0002-7356-1779  # Nicolas Matentzoglu
+  - https://orcid.org/0000-0003-2699-3066  # Meghan A. Balk
+
+classes:
+  anatomical_entity: UBERON:0001062
+  length: PATO:0000122
+  taxon: OBI:0100026
+  trait_term: PATO:0000051
+
+relations:
+  characteristic_of: RO:0000052
+  part_of: BFO:0000050
+
+annotationProperties:
+  exact_synonym: oio:hasExactSynonym
+  xref: oio:hasDbXref
+  measured_in_taxon: xsd:string
+  not_measured_in_taxon: xsd:string
+  comment: rdfs:comment
+
+vars:
+  element: 'anatomical_entity'
+  from: 'anatomical_entity'
+  to: 'anatomical_entity'
+  size_trait: 'trait_term'
+  fovt: "xsd:string"  # FOVT id
+  taxa: "taxon"
+  your_comments: "xsd:string"  # comments, clarifications for users
+
+data_list_vars:
+  def_ref: "'xref'"  # reference(s) to the source(s) of definition
+
+name:
+  text: "%s length"
+  vars:
+    - element
+
+annotations:
+
+  - annotationProperty: exact_synonym
+    text: "%s length"
+    vars:
+      - element
+
+  - annotationProperty: xref
+    text: "%s"  # give FOVT IRI here
+    vars:
+      - fovt
+
+  - annotationProperty: measured_in_taxon
+    text: "%s"
+    vars:
+      - taxa
+
+  - annotationProperty: not_measured_in_taxon
+    text: "%s"
+    vars:
+      - taxa
+
+  - annotationProperty: comment
+    text: "%s"
+    vars:
+      - your_comments
+
+def:
+  text: "The length of a %s from %s to %s."
+  vars:
+    - element
+    - from
+    - to
+  annotations:
+    - annotationProperty: xref
+      value: def_ref
+
+equivalentTo:
+  text: "'length' and ('characteristic_of' some ('part_of' some %s))"
+    # if part of a bone [e.g., 'distalmost' ('part of' some humerus)]
+    # NOTE: Everything is part of itself.
```

Comment:
It turns out that `BFO:0000050 part of` is actually not reflexive. You can check it by opening `ro.owl` in Protege and checking the Characteristics' pane in the 'Object Properties' tab.

Reply:
should we use "characteristic of part of"?

Reply:
"characteristic of part of" would cover the whole and all parts

Reply:
ah, so what should I use for just part of it if that part of the bone (e.g., distal side of a talus) isn't in the ontology?

Reply:
For that case the solution here is ok! But you may have to create a pattern for "side" of "anatomy" if non exists. But for the case you just want to express: _some_ part of the thalamus (distal side or otherwise), the solution we already have is sufficient.

Reply:
and just to check, that solution is: "'length' and ('characteristic_of' some ('part_of' some %s))"? What is the implication that it is not reflexive?

Reply:
The only relevant consequence is that the length of the _whole_ would not be classified under this term! It would be a sibling.

Reply:
ah, so another option could be:
EQ: length and ('part of' some (position and ('characteristic of' some element))) [e.g., length and ('part of' some ('distal side' and ('characteristic of' some calcaneus))) 
SubClassOf: 'calcaneus size'

Reply:
No, it would have to be:

```
length and characteristic of some (calcaneus and (part of some distal side))
```

Or some such. i don't know whether part of is the correct relationship. To determine this you will have to ask in Uberon slack. Basically just "can i use `part of` to connect an anatomical entity to a region like 'distal side'?"

Reply:
based on how distal epiphyses are defined, it'd be: length and characteristic of some (in distal side some calcaneus)

do I ask UBERON or OBA?

Reply:
definitely OBA!

Reply:
I will add to the agenda to discuss. I am struggling to explain over chat.

Reply:
Sorry I misunderstood your question. Uberon. You need to make an issue on Uberon tracker asking for the term:

- (in distal side some calcaneus)

Linking to this issue here, and asking for advice on how to logically define it. You can say that the issue has emerged because @cmungall does not like the complex composition in OBA (not sure I share his views here, but lets roll with it). 

What should happen: someone saying: you can define `in distal side some calcaneus` using XYZ logical definition, then creating a dosdp pattern for it. You can then just add a row in the table for each such term you need. Ping me if you get grieve from @dosumis or @shawntanzk about the request (term proliferation argument).